src/main/scala/edu/ie3/powerFactory2psdm/reduce/GridModelReducer.scala

```diff
@@ -0,0 +1,191 @@
+      csvSep,
+      inputFolderPath,
+      namingStrategy,
+      reducedGridName
+    )
+    if (!outputDir.exists()) {
+      outputDir.mkdir()
+    }
+    val initEmptyFiles = false
+    val sink =
+      new CsvFileSink(
+        outputDir.getCanonicalPath,
+        namingStrategy,
+        initEmptyFiles,
+        csvSep
+      )
+    sink.persistJointGrid(reducedGrid)
+
+    // write out mapping from node to system participant in csv file
+    val mappingFileName = outputDir.getPath + "/node_participant_mapping.csv"
+    writeMapping(
+      mappingFileName,
+      reducedGrid.getSystemParticipants.getFixedFeedIns.asScala.toSet
+    )
+  }
+
+  /** Reduces a grid by eliminating all system participants of the grid and
+    * connecting a new one per node. Furthermore creates and writes mapping from
+    * node to the connected system participant. This is done to have a grid
+    * which we can map primary data to each node without any additional models
+    * in the grid that draw or generate power.
+    *
+    * @param csvSep
+    *   csv separator of the grid to reduce
+    * @param inputFolderPath
+    *   folder path of the input grid
+    * @param namingStrategy
+    *   naming strategy used in input grid
+    * @param reducedGridName
+    *   name for the reduced grid
+    * @param outputDir
+    *   directory for storing grid and system participant mapping
+    */
+  private[reduce] def reduceGrid(
+      csvSep: String,
+      inputFolderPath: String,
+      namingStrategy: FileNamingStrategy,
+      reducedGridName: String
+  ): JointGridContainer = {
+
+    val inputGrid =
+      IoUtils.parsePsdmGrid(
+        reducedGridName,
+        csvSep,
+        inputFolderPath,
+        namingStrategy
+      )
+    val rawGridElements = inputGrid.getRawGrid
+
+    // create a system participant for each node
+    val fixedFeedIns = createFixedFeedIns(rawGridElements)
+    val systemParticipants = new SystemParticipants(
+      Set.empty[BmInput].asJava,
+      Set.empty[ChpInput].asJava,
+      Set.empty[EvcsInput].asJava,
+      Set.empty[EvInput].asJava,
+      fixedFeedIns.asJava,
+      Set.empty[HpInput].asJava,
+      Set.empty[LoadInput].asJava,
+      Set.empty[PvInput].asJava,
+      Set.empty[StorageInput].asJava,
+      Set.empty[WecInput].asJava
+    )
+
+    new JointGridContainer(
+      reducedGridName,
+      rawGridElements,
+      systemParticipants,
+      inputGrid.getGraphics
+    )
+
+  }
+
+  private def createFixedFeedIns(
+      gridElements: RawGridElements
+  ): Set[FixedFeedInInput] = {
+    val nodes = gridElements.getNodes
+    nodes.asScala.map(createFixedFeedIn).toSet
+  }
+
+  private def createFixedFeedIn(node: NodeInput): FixedFeedInInput = {
+    new FixedFeedInInput(
+      UUID.randomUUID(),
+      s"Participant-Node-${node.getUuid}",
+      node,
+      ReactivePowerCharacteristic.parse(
+        s"cosPhiFixed:{(0.0, 0.95)}"
+      ),
+      1.0.asKiloWatt,
+      0.95
```

Comment:
I only create these system participants so I can overwrite their behavior with primary data so the parameters don't matter and I do not think that this comes in handy anywhere else.

Reply:
Okay, in that scenario it actually works. However, the model after that is then "implicitly" bound to only be used in conjunction with a primary data source.

I'm a bit worrying about other people coming accross the repository in some months or years and using the created models in a fashion, that is physically / simulation-wise not sensible. Maybay you can add a warning to the README.md then?

Reply:
I did add words of warnings to the method descriptions. Since the models don't ship with the code you shouldn't be able to use the models incorrectly and I think with the added comment it should be sufficiently clear.

Reply:
I did add words of warnings to the method descriptions. Since the models don't ship with the code you shouldn't be able to use the models incorrectly and I think with the added comment it should be sufficiently clear.